setup.py

```diff
@@ -27,13 +27,38 @@
     "rsa>=3.1.4,<5",
 )
 
+# Unit test requirements.
+testing_extra_require = [
+    "flask",
+    "freezegun",
+    "mock",
+    "oauth2client",
+    "pyopenssl",
+    "pytest",
+    "pytest-cov",
+    "pytest-localserver",
+    "pyu2f",
+    "pyjwt",
+    "requests",
+    "urllib3",
+    "cryptography < 39.0.0",
+    "responses",
+    "grpcio",
+    # Async Dependencies
+    "pytest-asyncio; python_version > '3.0'",
+    "aioresponses; python_version > '3.0'",
+    "asynctest; python_version > '3.0'",
+    "aiohttp < 3.10.0; python_version > '3.0'",
+]
+
 extras = {
     "aiohttp": ["aiohttp >= 3.6.2, < 4.0.0.dev0", "requests >= 2.20.0, < 3.0.0.dev0"],
     "pyopenssl": ["pyopenssl>=20.0.0", "cryptography>=38.0.3"],
     "requests": "requests >= 2.20.0, < 3.0.0.dev0",
     "reauth": "pyu2f>=0.1.5",
     "enterprise_cert": ["cryptography", "pyopenssl"],
     "pyjwt": ["pyjwt>=2.0", "cryptography>=38.0.3"],
+    "testing_extra_require": testing_extra_require,
```

Comment:
```suggestion
    "testing": testing_extra_require,
```

so that the install can be simpler `pip install google-auth[testing]`. This also matches `google-cloud-aiplatform`

https://github.com/search?q=org%3Agoogleapis+%22%5C%22testing%5C%22%22+path%3Asetup.py&type=code

Alternatively, we could use `tests` instead:

https://github.com/search?q=org%3Agoogleapis+%22%5C%22tests%5C%22%3A%22+path%3Asetup.py&type=code